tests/upgrade_path/test_upgrade_path.py

```diff
@@ -133,17 +145,65 @@ def get_reboot_type(duthost):
         reboot_type = "warm-reboot"
     return reboot_type
 
+
 def check_sonic_version(duthost, target_version):
     current_version = duthost.image_facts()['ansible_facts']['ansible_image_facts']['current']
     assert current_version == target_version, \
         "Upgrade sonic failed: target={} current={}".format(target_version, current_version)
 
 
-def install_sonic(duthost, image_url):
-    res = duthost.reduce_and_add_sonic_images(new_image_url=image_url)
+def install_sonic(duthost, image_url, tbinfo):
+    mg_gwaddr = duthost.get_extended_minigraph_facts(tbinfo).get("minigraph_mgmt_interface", {}).get("gwaddr")
```

Comment:
Is this needed for the scenario that `image_url` is not a url but a path?

Reply:
Not needed. I will move this inside the check for HTTP/HTTPS URLs.